packages/cozy-stack-client/src/FileCollection.js

```diff
@@ -253,6 +253,37 @@ class FileCollection extends DocumentCollection {
     return resp.links.related
   }
 
+  /**
+   * Checks if the file belongs to the parent's hierarchy.
+   *
+   * @param  {string}  childFileID  The file ID of the child
+   * @param  {string}  childDirID   The dirID of the child
+   * @param  {string}  childPath    The child path
+   * @param  {string}  parentID     The parent's document id
+   * @return {boolean}              Whether the file is a parent's child
+   */
+  async isChild(childFileID, childDirID, childPath, parentID) {
+    if (childFileID === parentID || childDirID === parentID) {
+      return true
+    }
+    const paths = childPath.split('/')
```

Comment:
It may be better to use a method like path#dirname in a loop rather than a manual split. I don't want to miss some complex cases

Reply:
Ok with that, but what do you see as potential complex cases?

Reply:
I don't know (and that's why I would prefer delegating that to a lib).